packages/fast-components-react-msft/src/button/button.tsx

```diff
@@ -52,12 +61,40 @@ class Button extends Foundation<IButtonHandledProps & IManagedClasses<IMSFTButto
         }
     }
 
-    private generateInnerContent(): React.ReactElement<HTMLSpanElement> | (React.ReactNode | React.ReactNode[]) {
-        if (this.props.appearance === ButtonAppearance.lightweight || this.props.appearance === ButtonAppearance.justified) {
-            return <span className={get(this.props, "managedClasses.button_span")}>{this.props.children}</span>;
+    /**
+     * Renders slotted children in the appropriate slot
+     */
+    private renderChildrenBySlot(slot: ButtonSlot): JSX.Element {
+        if (Array.isArray(this.props.children)) {
+            return this.props.children.map((child: any, index: number) => {
```

Comment:
I think we can use the `React.Children` instead and then you can skip the `isArray` check.